### ISSUE TITLE
[NAOT][Maui] iOS Maui sample app startup getting slower

### ISSUE THREAD

**Issue by matouskozak** [dotnet/runtime#104083]
# Description
MAUI iOS sample app startup measurements are reporting incrementally slower times when run with NAOT backend while NAOT HelloWorld measurements are stable. Also the same measurements with Mono backend don't report any increases in startup times.

![image](https://github.com/dotnet/runtime/assets/55735845/7e47559e-b8c4-4f48-8c65-120588da6040)

**Comment by agocke** [comment]
How do we go about investigating this?

**Comment by matouskozak** [comment]
> > How do we go about investigating this?
> 
> Is it easy enough to re-measure these? Say, try .NET 9 RTM vs .NET 9 Preview 3 vs .NET 9 Preview 1 on the same phone with the same OS version? We haven't really done any work affecting the startup path on the runtime side and if this is not accompanied with steady size regression (we do have some things where worse size could correlate to worse startup), I don't see any explanation for a steady decline. Was there any work done on the MAUI side that could explain this?

We should be able to re-measure this locally to verify. 

To put more information to the issue, we currently see that both NativeAOT and Mono MAUI backends are slowly regressing as per ![Image](https://github.com/user-attachments/assets/2409ca86-0a38-4a25-9842-d317fa25d406)

PowerBI only offers us 3 months of data, we might be able to get more information from Kusto.

**Comment by MichalStrehovsky** [comment]
> To put more information to the issue, we currently see that both NativeAOT and Mono MAUI backends are slowly regressing as per

Is this measuring .NET 9 like the chart legend says, or does it measure main?

**Comment by matouskozak** [comment]
> > To put more information to the issue, we currently see that both NativeAOT and Mono MAUI backends are slowly regressing as per
> 
> Is this measuring .NET 9 like the chart legend says, or does it measure main?

It should be the main (I think it didn't get updated after GA), I'll check with the perf team to confirm.

**Comment by ivanpovazan** [comment]
> We should be able to re-measure this locally to verify.

Agreed, we can rerun the dotnet perf maui scenarios locally with different versions of the runtime as Michal suggested.
If we confirm the regression, we can further analyze where the slowdown comes from by using Xcode Instruments to profile the app startup and compare the profiles.

Let me know if I can help with this.

**Comment by agocke** [comment]
I think we should try to isolate MAUI and Native AOT/the core framework. If I understand this benchmark correctly, it's pulling bits from the latest MAUI, along with the latest runtime bits. So any regression here could be anywhere up or down the stack. It's going to be very hard to isolate the cause of any particular regression with that setup. I think we should at least have another MAUI benchmark that freezes the version of .NET. That should give us a sense of where to investigate.
